Standards/scs-xxxx-v1-ipv4-networking-standard.md

```diff
@@ -0,0 +1,79 @@
+
+## Motivation
+
+The motivation behind establishing this standard is to enhance interoperability, improve
+security measures, and streamline the operational processes across different SCS clouds.
+It addresses the need for a unified procedure in handling IPv4 networking to facilitate
+ease of use for IaaS users and DevOps teams.
+
+## Design Considerations
+
+### Options considered
+
+#### _Neutron Routers_
+
+Usage of Neutron Routers: To manage traffic between internal and external networks Neutron Routers **MUST** be used as the default gateway for VMs requiring access to external networks and the internet, thereby facilitating the routing of traffic and enhancing network security.
+
+CSPs **SHOULD** use OVN or L3agent as High Availability (HA) service deployments.
+Standard external networks **MUST NOT** be made accessible as _shared networks_. It is advised that external networks are only reachable by the usage of routing and floating IPs.
+However, for special use cases like certain storage or VPN solutions it could be useful to allow _direct access networks_.
+
+External networks and subnets **SHOULD** (very strong should) be configured with _--no-dhcp_ (DHCP - Dynamic Host Configuration Protocol). It is more secure to configure it like this, since it gives less space for reflection attacks, e.g. _Denial of Service_ (DOS) attacks. If _dhcp_ is configured, certain firewall configurations **MUST** be made to catch IPs from the _Neutron dhcp agent_ in the public network.
+
+#### _Neutron Plugins_
+
+Neutron Plugins: A SCS conform CSP **MAY** use RBAC and VPNaaS plugins.
+(Neutron RBAC needs to be configured explicitly to be able to use it. If configured, Neutron configurations can be shared across OpenStack projects. It also can be beneficial for admins, since an admin could bind external networks only to certain projects.)
+
+#### _Security Groups_
+
+Security Group Policies: Standardized security group policies **SHOULD** be applied to all instances utilizing public IPv4 addresses. These policies must define and enforce access
+controls to ensure the security of the cloud environment.
+Security Groups **SHOULD** be enabled by default but **MUST** be capable of being switched off.
+
+#### _Quota & Monitoring_
+
+Quota: The standard quota of floating IPs and routers **SHOULD** be rather small, e.g. 3-5 floating IPs. This ensures a more fair distribution of these resources for all cloud users. If a user wants to use more of these resources, the user **SHOULD** be able to pay for more.
+
+IP Usage Monitoring: SCS CSPs **SHOULD** implement monitoring solutions to track the utilization of IPv4 addresses. This facilitates efficient management of resources and supports capacity planning efforts.
+
+#### _External Network Naming_
+
+All SCS clouds **SHOULD** adopt the naming convention
+scs-external-net for external networks. This standardization facilitates easier identification and management of external network resources.
```

Comment:
We have to clarify why we would need a Naming convention at all. Renaming networks without good reasons is not really helpful.

Pre-statement: external networks can be listed with `openstack network list --external`.

1. How many CSPs have more than 1 external network (for IPv4) ?
2. How many CSPs have multiple different subnets for one external network == more than one floating-IP-pool ?
3. Are there external networks that are only for one specific customer (i have seen something like this)?

Pro:
- distinguish between IPv4 and IPv6 external networks

Con:
- how to deal with multiple IPv4 external networks?
- use other options like tags:
```
stack@devstack:~/devstack$ openstack network show public
+---------------------------+----------------------------------------------------------------------------+
| Field                     | Value                                                                      |
+---------------------------+----------------------------------------------------------------------------+
| admin_state_up            | UP                                                                         |
| availability_zone_hints   |                                                                            |
| availability_zones        |                                                                            |
| created_at                | 2024-01-24T16:12:31Z                                                       |
| description               |                                                                            |
| dns_domain                | None                                                                       |
| id                        | 73edb86b-d7ab-4db3-82b7-25fa8b012e40                                       |
| ipv4_address_scope        | None                                                                       |
| ipv6_address_scope        | None                                                                       |
| is_default                | True                                                                       |
| is_vlan_transparent       | None                                                                       |
| mtu                       | 1500                                                                       |
| name                      | public                                                                     |
| port_security_enabled     | True                                                                       |
| project_id                | 15f2ab0eaa5b4372b759bde609e86224                                           |
| provider:network_type     | flat                                                                       |
| provider:physical_network | public                                                                     |
| provider:segmentation_id  | None                                                                       |
| qos_policy_id             | None                                                                       |
| revision_number           | 4                                                                          |
| router:external           | External                                                                   |
| segments                  | None                                                                       |
| shared                    | False                                                                      |
| status                    | ACTIVE                                                                     |
| subnets                   | 3e0206bc-53c8-44ca-a0f1-2c2548bba766, 84dffd43-6d7f-4c2f-9180-8f0f0b83c9d4 |
| tags                      | IPv4                                                                       |
| tenant_id                 | 15f2ab0eaa5b4372b759bde609e86224                                           |
| updated_at                | 2024-03-28T09:39:03Z                                                       |
+---------------------------+----------------------------------------------------------------------------+
stack@devstack:~/devstack$ openstack network list --external --long --tag IPv4
+----------------------------+--------+--------+----------------------------+-------+--------+----------------------------+--------------+-------------+--------------------+------+
| ID                         | Name   | Status | Project                    | State | Shared | Subnets                    | Network Type | Router Type | Availability Zones | Tags |
+----------------------------+--------+--------+----------------------------+-------+--------+----------------------------+--------------+-------------+--------------------+------+
| 73edb86b-d7ab-4db3-82b7-   | public | ACTIVE | 15f2ab0eaa5b4372b759bde609 | UP    | False  | 3e0206bc-53c8-44ca-a0f1-   | flat         | External    |                    | IPv4 |
| 25fa8b012e40               |        |        | e86224                     |       |        | 2c2548bba766, 84dffd43-    |              |             |                    |      |
|                            |        |        |                            |       |        | 6d7f-4c2f-9180-            |              |             |                    |      |
|                            |        |        |                            |       |        | 8f0f0b83c9d4               |              |             |                    |      |
+----------------------------+--------+--------+----------------------------+-------+--------+----------------------------+--------------+-------------+--------------------+------+
```

I am pretty much for investigating those tags! Help from CSPs is wanted (we don't want to accidentally render a network not working anymore :D )